src/sentry/replays/post_process.py

```diff
@@ -190,6 +220,32 @@ def generate_sorted_urls(url_groups: list[tuple[int, list[str]]]) -> Iterator[st
         yield from url_group
 
 
+def generate_viewed_by_response(
+    replay_id: str,
+    viewed_by_ids: list[int],
+    actor: User,
+) -> ReplayViewedByResponse:
+    """Fetch user objects from postgres, serialize them, and format to the output expected by ReplayViewedByEndpoint."""
+    global user_serializer
+
+    # TODO: what happens if a viewed_by_id is invalid or unauthorized?
```

Comment:
how could it be unauthorized?

Reply:
Didn't get to the bottom of it but I suspected, from the as_user param of serialize_many, that some kind of authorization is needed for some user objects.